client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoResultSet.java

```diff
@@ -183,7 +183,9 @@ public AsyncIterator(Iterator<T> dataIterator, StatementClient client)
                     }
                 }
                 catch (InterruptedException e) {
-                    handleInterrupt(e);
+                    client.close();
+                    rowQueue.clear();
```

Comment:
why does cleaning rowQueue here matter?

Reply:
here it is probably not important. it's more like "why not?"

Reply:
discussed offline and agreed to keep